sphinx/ext/autodoc/importer.py

```diff
@@ -198,7 +198,8 @@ def get_object_members(
             if name not in members:
                 members[name] = Attribute(name, True, value)
 
-        superclass = subject.__mro__[1]
+        # enumerations are created as `EnumName([mixin_type, ...] [data_type,] enum_type)`
+        superclass = subject.__mro__[-2]
```

Comment:
Perhaps also note that the superclass that we want is the data_type class. Can there be multiple data_type classes?
```suggestion
        # enumerations are created as ``EnumName([mixin_type, ...] [data_type,] enum_type)``
        superclass = subject.__mro__[-2]
```

Reply:
According to Python, no: [https://docs.python.org/3/howto/enum.html#restricted-enum-subclassing](https://docs.python.org/3/howto/enum.html#restricted-enum-subclassing):

> A new [Enum](https://docs.python.org/3/library/enum.html#enum.Enum) class must have one base enum class, up to one concrete data type, and as many [object](https://docs.python.org/3/library/functions.html#object)-based mixin classes as needed. 

The order of these base classes is:

```python
class EnumName([mix-in, ...,] [data-type,] base-enum):
    pass
```

However, thank you for pointing it since it also told me that it's possible to have *no* data-type class. In this case, the code still won't work properly. According to [https://docs.python.org/3/howto/enum.html#others](https://docs.python.org/3/howto/enum.html#others), a *data-type* class is a mixin class which defines a `__new__`, so I should technically check if this is the case or not which I will do this week-end.

Reply:
Actually, after re-reading the issue, what we wanted is not the data-type but the base enumeration type. The reason is that we want to exclude enum-related methods and keep only those from the mixin and possibly the data-type class.

Reply:
After some thoughts, here is what I decided to do *for this specific PR*:

- The original issue stems from the fact that some enumeration methods are not filtered out, like `_generate_next_value` (or smth like that).  As such, the functionality that we want is to be able to remove those private methods. 
- The way we achieved it before is to take `superclass = cls.__mro__[1]`, because we assumed that the class is of the form `class MyEnum(Enum)` leading to mro of `[MyEnum, Enum, object]`. 
- This caused the following issue: if we declare `class MyEnum(str, Enum)`, we end up with filtering out methods from `str` and not `Enum`. 

Now, if we use `cls.__mro__[-2]`, we will always target `Enum`. Even when subclassing another enum, e.g., `class Other(MyEnum)`, the MRO would still be `[Other, MyEnum, Enum, object]`. What we want is to also be able to keep methods from `MyEnum` in `Other`, so we don't want to actually exclude `MyEnum` methods, only those coming from `Enum`.

So we can actually directly use `Enum`. Now, this PR also solves the other issue that arose from the original issue. The other issue is that `dir(MyEnum)` does *not* return public methods or special ones, so, some methods are not rendered when inheriting from a mixin + enum type. However, that's another question. Some are exposed because they are indeed no in `Enum.__dict__` and that's fine, but at the same time, those that have same names as in `Enum.__dict__` such as special methods are still *not* rendered. Now, if `special-members` option is used, the special members will be included correctly (before, even with special-members option, they were not included because they were never stored).